src/Microsoft.Data.Sqlite/SqliteCommand.cs

```diff
@@ -111,11 +114,16 @@ public new virtual SqliteDataReader ExecuteReader(CommandBehavior behavior)
             do
             {
                 Sqlite3StmtHandle stmt;
-                var rc = NativeMethods.sqlite3_prepare_v2(
-                    Connection.DbHandle,
-                    tail,
-                    out stmt,
-                    out tail);
+                var rc = NativeMethods.sqlite3_prepare_v2_blocking(
+                        Connection.DbHandle,
+                        tail,
+                        out stmt,
+                        out tail,
+                        CommandTimeout*1000);
```

Comment:
This is cleaner than I thought it would be. Only one command ever runs on the connection at a time, so the fact that the timeout is actually associated with the connection doesn't ever really mater.